agent/connect/ca/provider_vault.go

```diff
@@ -864,6 +864,36 @@ func (v *VaultProvider) setNamespace(namespace string) func() {
 	}
 }
 
+// autotidyIssuers sets Vault's auto-tidy to remove expired issuers
+// Returns a boolean on success for testing (as there is no post-facto way of
+// checking if it is set). Logs warnings on failure to set and why, returning the
+// log message for test purposes as well.
+func (v *VaultProvider) autotidyIssuers(path string) (bool, string) {
+	s, err := v.client.Logical().Write(path+"/config/auto-tidy",
```

Comment:
We need to update our Vault ACL docs to include this new convenience permission that's needed (on `config/auto-tidy`).

- https://developer.hashicorp.com/consul/docs/connect/ca/vault#configure-vault-acl-policies
- https://developer.hashicorp.com/consul/tutorials/vault-secure/vault-pki-consul-connect-ca#create-vault-policies

Reply:
We might want to make it clear that this permission is recommended, but optional. If not provided, they should configure tidy-expired-issuers themselves on Vault 1.13+, or manually tidy the expired issuers themselves as the number of issuers grows large (100+).

Reply:
Do we want to include the documentation update in this PR?

Reply:
I'm going to assume yes and start looking at the docs changes.

Reply:
I think that's best (so we don't forget about it). Though I'd also understand if you wanted to make that a separate PR.

Reply:
The policy in the existing documentation already give the required access to the auto-tidy settings. So no changes to docs are needed.